soc/riscv32/riscv-privilege/sifive-freedom/soc.h

```diff
@@ -40,7 +41,11 @@
 #define SIFIVE_BACKUP_REG_BASE	     0x10000080
 
 /* lib-c hooks required RAM defined variables */
-#define RISCV_RAM_BASE               CONFIG_RISCV_RAM_BASE_ADDR
-#define RISCV_RAM_SIZE               CONFIG_RISCV_RAM_SIZE
+#define RISCV_RAM_BASE               DT_SIFIVE_DTIM0_0_BASE_ADDRESS
+#define RISCV_RAM_SIZE               DT_SIFIVE_DTIM0_0_SIZE
+
+/* Flash defined variables for linker */
+#define RISCV_ROM_BASE		DT_SOC_NV_FLASH_0_BASE_ADDRESS
+#define RISCV_ROM_SIZE		DT_SOC_NV_FLASH_0_SIZE
```

Comment:
Instead of hard-coding the device we expect for RAM and ROM, we should use the chosen node in the devicetree.
```
chosen {
		zephyr,sram = &dtim;
		zephyr,flash = &flash0;
};
```

Reply:
Especially since other SiFive SoCs won't necessarily have a DTIM or soc-nv-flash.

Reply:
I changed it to use `DT_SRAM` and `DT_FLASH` symbols. Is it ok to assume that both ROM and RAM is present or should it be ifdefed somehow (e.g., by using `DT_CHOSEN_ZEPHYR_FLASH`)?

Reply:
Awesome, perfect!

I'm *pretty sure* this is sufficient, since [include/arch/riscv32/common/linker.ld](https://github.com/zephyrproject-rtos/zephyr/tree/master/include/arch/riscv32/common/linker.ld) ignores `CONFIG_RISCV_ROM_BASE_ADDR` if `CONFIG_XIP=n`.